### DISCUSSION TITLE
This is a question about how to construct Voronoi based on a boundary

### DISCUSSION THREAD

**Discussion by Wayne8887** [gama-platform/gama#270] [Q&A]
![微信图片_20240727120825](https://github.com/user-attachments/assets/f294f494-61f8-45d0-91c9-19cdbddcc1b3)
This is a tutorial on how to use Voronoi on the official website

Now I have made a Tyson polygon according to the instructions on the official website.
code：list<geometry> var0 <- voronoi([{10,10},{50,50},{90,90},{10,90},{90,10}]);

![微信图片_20240727121330](https://github.com/user-attachments/assets/c23a2f69-4d37-4367-af8d-07134b519821)
This is the output result

But when I input the interface I want, such as using the example code provided by the official website.
list<geometry> var1 <- voronoi([{10,10},{50,50},{90,90},{10,90},{90,10}], square(300));
But there was no result displayed
![微信图片_20240727121607](https://github.com/user-attachments/assets/6ecc4652-7f25-4ba2-8941-3dae98de62fe)

Here is my code
![Uploading 微信图片_20240727121646.png…]()

**Comment by Wayne8887** [comment]
Sincerely, may I ask how teachers can solve this problem

**Comment by lesquoyb** [comment]
I think you just forgot to draw the result of the voronoi operator, here is one example:
```
model voronoi

global {
	list<geometry> points <- [{10,10},{50,50},{90,90},{10,90},{90,10}];
	list<geometry> clusters <- voronoi(points);
	init {
		write clusters;
	}
}


experiment a {
	
	output {
		display scene {
			graphics "points and clusters"{
				loop p over:points{
					draw p color:#black;
				}
				loop c over:clusters{
					draw c color:#red wireframe:true;
				}
			}
		}
	}
}
```
